### PR TITLE
feat(modelarts): add a new data source to get services

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccDatasourceModelartServices_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts -v

=== RUN   TestAccDatasourceModelartServices_basic
=== PAUSE TestAccDatasourceModelartServices_basic
=== CONT  TestAccDatasourceModelartServices_basic
--- PASS: TestAccDatasourceModelartServices_basic (111.81s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts	111.902s

```

### CODE REVIEW

File: docs/data-sources/modelarts_services.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+* `owner` - User to which a service belongs.
+
+<a name="ModelartServices_ServicesSchedule"></a>
+The `schedule` block supports:
+
+* `duration` - Value mapping a time unit.  
+  For example, if the task stops after two hours, set time_unit to HOURS and duration to 2.
+
+* `time_unit` - Scheduling time unit. Possible values are DAYS, HOURS, and MINUTES.
```
Comment (houpeng80) [inline docs/data-sources/modelarts_services.md]:
```suggestion
* `time_unit` - Scheduling time unit. Possible values are **DAYS**, **HOUR**S, and **MINUTES**.
```

Comment (hmbig2) [reply docs/data-sources/modelarts_services.md]: done

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_services.go [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+		})
+	}
+	return rst
+}
+
+func flattenServicesAdditionalProperty(resp interface{}) []interface{} {
+	var rst []interface{}
+	curJson, err := jmespath.Search("additional_properties", resp)
+	if err != nil {
+		log.Printf("[ERROR] error parsing additional_properties from response= %#v", resp)
```
Comment (ShiChangkuo) [inline huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_services.go]: should add the strings of err

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_services.go [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+		},
+	}
+	return rst
+}
+
+func flattenServicesAdditionalPropertySmnNotification(resp interface{}) []interface{} {
+	var rst []interface{}
+	curJson, err := jmespath.Search("smn_notification", resp)
+	if err != nil {
+		log.Printf("[ERROR] error parsing smn_notification from response= %#v", resp)
```
Comment (ShiChangkuo) [inline huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_services.go]: ditto

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_services.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_services.go]: done

File: huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_services.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/modelarts/data_source_huaweicloud_modelarts_services.go]: done

**Pull request merged**